devito/finite_differences/derivative.py

```diff
@@ -129,17 +169,37 @@ def __new__(cls, expr, *dims, **kwargs):
         # of the derivative
         variable_count = [sympy.Tuple(s, new_dims.count(s))
                           for s in filter_ordered(new_dims)]
-
-        # Construct the actual Derivative object
-        obj = Differentiable.__new__(cls, expr, *variable_count)
-        obj._dims = tuple(OrderedDict.fromkeys(new_dims))
-        obj._fd_order = fd_orders
-        obj._deriv_order = orders
-        obj._side = kwargs.get("side")
-        obj._transpose = kwargs.get("transpose", direct)
-        obj._subs = as_tuple(kwargs.get("subs"))
-        obj._x0 = kwargs.get('x0', None)
-        return obj
+        return new_dims, orders, fd_orders, variable_count
+
+    def __call__(self, x0=None, fd_order=None, side=None):
+        if self.ndims == 1:
+            _fd_order = fd_order or self._fd_order
+            _side = side or self._side
+            new_x0 = {self.dims[0]: x0} if x0 is not None else self.x0
+            return self._new_from_self(fd_order=_fd_order, side=_side, x0=new_x0)
+
+        if side is not None:
+            raise TypeError("Side only supported for first order single"
+                            "Dimension derivative such as `.dxl` or .dx(side=left)")
+        # Cross derivative
+        new_x0 = self._x0 or {}
+        _fd_order = self.fd_order._getters
```

Comment:
watch out: this is changing state because you're *updating* (line below) `self.fd_order._getters` in-place